### DISCUSSION TITLE
Unable to access all the elements through material object

### DISCUSSION THREAD

**Discussion by rayaprolu143** [idaholab/moose#21130] [Q&A General] [answered]
Hi,

Using the `FeatureVolumeVectorPostProcessor`, I obtained the element IDs belonging to the features (e.g. precipitate, bubble). Now, I want to assign special properties to the elements belonging to the feature. However, through the material object, I cannot access the element IDs of all the elements in the domain when the code runs on multiple processors, but I can access all the elements when the code runs in serial. Here, I am attaching the material object file and the input file. 

If you play with the file, you will notice that only a subset of element ids listed in the root node will be printed out when the code runs in parallel mode, whereas all the element ids will be printed when the code runs serially. When I tried printing the rank of the processor, it only printed the root node's rank, '0'. I want to access all the elements of the domain through the material object, which I do not know how to achieve. 
[MaterialObject.C.txt](https://github.com/idaholab/moose/files/8770756/MaterialObject.C.txt)
[MaterialObject.h.txt](https://github.com/idaholab/moose/files/8770757/MaterialObject.h.txt)
[Sample_input_file.i.txt](https://github.com/idaholab/moose/files/8770758/Sample_input_file.i.txt)

**Comment by rayaprolu143** [comment, marked as answer]
I figured out that only the root node can execute the print statement. There is no problem.

**Reply by GiudGiud** [reply]
add `--keep-cout` to the command line call to have console output for all ranks

**Reply by rayaprolu143** [reply]
Thanks a lot!
